src/rez/plugin_managers.py

```diff
@@ -52,18 +53,23 @@ def extend_path(path, name):
     init_py = "__init__" + os.extsep + "py"
     path = path[:]
 
-    for dir_ in config.plugin_path:
-        if not os.path.isdir(dir_):
-            if config.debug("plugins"):
-                print_debug("skipped nonexistant rez plugin path: %s" % dir_)
-            continue
+    def append_if_valid(dir_):
+        if os.path.isdir(dir_):
+            subdir = os.path.normcase(os.path.join(dir_, pname))
+            initfile = os.path.join(subdir, init_py)
+            if subdir not in path and os.path.isfile(initfile):
+                path.append(subdir)
 
-        subdir = os.path.join(dir_, pname)
-        # XXX This may still add duplicate entries to path on
-        # case-insensitive filesystems
-        initfile = os.path.join(subdir, init_py)
-        if subdir not in path and os.path.isfile(initfile):
-            path.append(subdir)
+        elif config.debug("plugins"):
+            print_debug("skipped nonexistant rez plugin path: %s" % dir_)
+
+    # Extend old-style plugins
+    for dir_ in config.plugin_path:
+        append_if_valid(dir_)
+    # Extend new-style plugins
+    for loader, name, ispkg in pkgutil.iter_modules():
```

Comment:
There is a problem here we need to address I think. I'm already a bit nervous about iterating over all modules (from a performance pov), but this is actually causing iter_modules to be called once *per plugin type*. We need to do something here to avoid that. I think a simple module list cache would do the trick.

Reply:
Yeah, a module path list getter function with cache in instance of `RezPluginManager` should be it.

Reply:
implemented in 2b9cd2b